### DISCUSSION TITLE
Basic question about caching and image IDs

### DISCUSSION THREAD

**Discussion by jordanc-jrj** [docker/buildx#1779] [Q&A]
I'm trying to gain a better understanding of the buildx build process, caching, and docker image IDs.  I've put together a simple example where I:

1. Create a bake file for a single image with a local `cache-from` and `cache-to` location
2. Create a builder using the `docker-container driver`
3. Run the build (1), and save off the `image inspect` output.
4. Re-run the build (2), and save off the `image inspect` output.
5. Delete/recreate the builder, but leave the local cache.
6. Re-run the build (3), and save off the `image inspect` output.

The results I'm seeing are that the `inspect` outputs and image IDs from build 1 and 2 match completely, which makes sense to me.  The `inspect` result from build 3 matches completely, except the `Id` sha itself is different.  Everything else in the output matches.  Is this to be expected when the builder is recreated?  Looking for insight as to what would cause the ID to change?

Steps to reproduce:
1. Clone the repo: https://github.com/jordanc-jrj/docker-buildx-demo
2. Execute: `./run.sh`
3. Compare the inspect output files in the `./inspect` directory

Here's my `docker info` output for reference:
```
> docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.4
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.17.3
    Path:     /usr/libexec/docker/cli-plugins/docker-compose
  scan: Docker Scan (Docker Inc.)
    Version:  v0.23.0
    Path:     /usr/libexec/docker/cli-plugins/docker-scan

Server:
 Containers: 9
  Running: 4
  Paused: 0
  Stopped: 5
 Images: 111
 Server Version: 23.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 nvidia runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2806fc1057397dbaeefbea0e4e17bddfbd388f38
 runc version: v1.1.5-0-gf19387a
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
 Kernel Version: 5.14.0-1059-oem
 Operating System: Ubuntu 20.04.5 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 31.07GiB
 Name: or-lin-1fh
 ID: Q5U7:XPBG:LIUG:GDNI:5TST:B7CL:4M5L:UZ5H:TUQW:P76V:VYWH:S2IR
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  localhost:5000
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by tonistiigi** [comment]
You can look up the actual image config in `/var/lib/docker/image/overlay2/imagedb/content/sha256/` and see what the difference is. If it is a timestamp you can fix it with `SOURCE_DATE_EPOCH` but probably the layers are different as well as you create new files in `RUN` that store their creation time.

**Reply by jordanc-jrj** [reply]
Thanks. I see in my example file, its the `WORKDIR` timestamp that differs between the builds.  What is the reason that this only happens when the builder container is recreated, even though the cache store stays the same?
